### PR TITLE
LPS-196289 - Send prepopulated filters

### PR DESCRIPTION
**Pull request by juanjofgliferay**
Story https://liferay.atlassian.net/browse/LPS-185991
Task https://liferay.atlassian.net/browse/LPS-196289

## Problem
After defining a date filter in the Dataset manager it does not show it in the fragment. It also happens with selection filter.

## Solution
We need to define a filter and cover some use cases:
- if the user does not select any date, the fragment does not show any filter applied
- if the user selects at least one date, the fragment will display the filter applied
- if the user defines a selection filter that uses preselected values, the filter with the preselected values appears in the fragment

## Other changes
- fix url for `date` filter so the user can delete them
- refactor/rename `fieldName` dropdown component (soon replaced by Clay Picker)
- update persistance layer: `name` by `label`

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx [hunk 1]
```diff
@@ -271,13 +269,13 @@ function AddFDSFilterModalContent({
 		fieldNames?.includes(item.name) ? item.name : undefined
 	);
 
-	const CellRendererDropdown = ({
-		cellRenderers,
+	const FieldNameDropdown = ({
+		fields,
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx:273]: Nice catch :tada:

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 2]
```diff
@@ -437,16 +437,55 @@ private JSONArray _getFiltersJSONArray(
...
+					).put(
 						"entityFieldType", FDSEntityFieldTypes.DATE
 					).put(
 						"id", properties.get("fieldName")
 					).put(
 						"label", properties.get("name")
 					).put(
-						"max", _getDateJSONObject(properties.get("to"))
-					).put(
-						"min", _getDateJSONObject(properties.get("from"))
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:449]: @dsanz `min` and `max` are actually missing from UI. Should we add them in Q4?

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 3]
```diff
@@ -437,16 +437,55 @@ private JSONArray _getFiltersJSONArray(
...
+
+					String from = MapUtil.getString(properties, "from");
+					String to = MapUtil.getString(properties, "to");
+
+					if (Validator.isNotNull(to) || Validator.isNotNull(from)) {
+						hasPreloadedData = true;
+					}
+					else {
+						hasPreloadedData = false;
+					}
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:450]: We can simplify this to 
```java
boolean hasPreloadedData = Validator.isNotNull(to) || Validator.isNotNull(from);
```

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx [hunk 4]
```diff
@@ -271,13 +269,13 @@ function AddFDSFilterModalContent({
 		fieldNames?.includes(item.name) ? item.name : undefined
 	);
 
-	const CellRendererDropdown = ({
-		cellRenderers,
+	const FieldNameDropdown = ({
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx:272]: This is close candidate be turned into a app-level component that can be used both for cell renderers and for filters. I'm more than fine to have them separated for now, just note it down for future evaluation

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx [hunk 5]
```diff
@@ -920,7 +922,7 @@ function Filters({fdsView, namespace}: IFDSViewSectionInterface) {
 						processClose();
 
 						const url = `${
-							item.type === 'date-time'
+							item.type === 'date' || item.type === 'date-time'
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx:925]: If we focus on `date`  fields for the moment, keeping `date-time` here may be confusing for out future selves. It does not hurt anyways

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 6]
```diff
@@ -437,16 +437,55 @@ private JSONArray _getFiltersJSONArray(
...
+							return JSONUtil.put(
+								"from",
+								() -> {
+									if (Validator.isNull(from)) {
+										return null;
+									}
+
+									return _getDateJSONObject(
+										properties.get("from"));
+								}
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:476]: If we add to `_getDateJSONObject` something like 
```java
if (object == null) {
    return null;
}
```
we can simplify this to
```java
JSONUtil.put("from", () -> _getDateJSONObject(from))
```

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 7]
```diff
@@ -437,16 +437,55 @@ private JSONArray _getFiltersJSONArray(
 				String type = MapUtil.getString(properties, "type");
 
 				if (Objects.equals(type, "date")) {
+					boolean hasPreloadedData;
+
+					String from = MapUtil.getString(properties, "from");
+					String to = MapUtil.getString(properties, "to");
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:443]: Do we need to convert this to string? For a `null` check, `Object from = ...` should is enough, and we can reuse the variable for `_getDateJSONObject(from)`

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx [hunk 8]
```diff
@@ -100,7 +100,7 @@ function AddFDSFilterModalContent({
 		boolean
 	>();
 	const [from, setFrom] = useState<string>(
-		(filter as IDateFilter)?.from ?? format(new Date(), 'yyyy-MM-dd')
+		(filter as IDateFilter)?.from ?? ''
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx:103]: Any reason to remove the defaults? I don't mind the blank default, just curious.

**Review by markocikos — COMMENTED** [review]
@juanjofgliferay mostly looks good, some minor stuff inline

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx (hunk 4 above)
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx:272]: It should be changed by Clay Picker soon, but the name was confusing

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java (hunk 2 above)
Comment (dsanz) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:449]: I think so, it looks as a nice-to-have. I've added https://liferay.atlassian.net/browse/LPS-197287 for further refinement @pablo-agulla

**Comment by dsanz** [comment]
Master got the changes needed to send this one :tada: @juanjofgliferay

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 9]
```diff
@@ -562,7 +562,7 @@ private JSONArray _getFiltersJSONArray(
 					).put(
 						"id", properties.get("fieldName")
 					).put(
-						"label", properties.get("name")
+						"label", properties.get("label")
```
Comment (izaera) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:565]: Not 100% sure, but I think this should be done like in here -> https://github.com/liferay-frontend/liferay-portal/pull/3683/commits/d4dfca99946e029275de9f86ce6dc644d277a014#diff-14ac39b8c7158c0601a7ccd2a2cacb7d24afe76cdd503e7f3c4921a11e3d22c9R518

@markocikos Am I right?  :thinking:

Comment (izaera) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:565]: Turns out _getValue was added by @juanjofgliferay , so I guess we don't need to ask @markocikos :sweat_smile:

Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:565]: @izaera, you're right, one more time. That's the case to get a  default label in the filter

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Review by markocikos — APPROVED** [review]
@juanjofgliferay I tested cases for all three filter types, found no issues. Code looks good. As far as I'm concerned, we can ship this!

**Comment by dsanz** [comment]
Yep I don't see anything concerning.

**Comment by dsanz** [comment]
all yours @mateusralv  :muscle:

**Comment by mateusralv** [comment]
Hi @juanjofgliferay and @dsanz I took a look and the failures are not related 🎉

[Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1695923557168&testrayRunIdA=2433182968&testrayRunIdB=2433358509)

I decided to do an exploratory test related to the filters and found: 
1. The client extension filter is shown in the fragment 🎉
2. I couldn't create a filter related to date range, see the video below.

https://github.com/liferay-frontend/liferay-portal/assets/44584050/a39a5f51-11b8-4204-ae1c-e9223fc76f23

**Comment by juanjofgliferay** [comment]
> [Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1695923557168&testrayRunIdA=2433182968&testrayRunIdB=2433358509)
> 
> I decided to do an exploratory test related to the filters and found:
> 
> 1. The client extension filter is shown in the fragment 🎉 
:muscle: 
> 2. I couldn't create a filter related to date range, see the video below.
To test a date range filter I usually create a custom object and add a new field of type date

**Comment by mateusralv** [comment]
True! It works! it didn't work on the ent point I tested because they had date-time fields.

 Thanks @juanjofgliferay safe to forward. 🚀 

image

**Comment by dsanz** [comment]
ci:forward:force

**Comment by dsanz** [comment]
ci:forward:force

**Pull request closed without merging**